### ISSUE TITLE
People are accidentally losing their wip edits by forgetting/missing to save

### ISSUE THREAD

**Issue by Powersource** [Edgeryders-Participio/realities#208]
Source https://forum.blivande.com/t/realties-suggestions-of-features/1710/5

Hugi said:

> DianaMO, hampus and I have been working in Realities today, and it has happened many times that DianaMO has lost long texts because she has forgotten to save. This tells me that the UX is not intuitive and will frustrate people - one of the destructive UX fails is when you lose work.
> 
> I see two possible ways through this:
> 
> * Save on input, as you type. Can probably also have unintended consequences.
> * Warn before you are about to navigate somewhere else, in case you have not saved.

**Comment by Powersource** [comment]
I think I'd prefer the second solution, since e.g. we've been talking about the ux around notifications, how would that work in the autosave case? I don't know how techincallly feasible this is though, there are some things built into browsers for this but I don't know how well it works in a SPA?

**Comment by aerugo** [comment]
@gustavlrsn has implemented a confirm to delete dream modal on Dreams, so you could look at that example

**Comment by Powersource** [comment]
Yeah I don't mean just modals if that's what you mean. But detection of navigating away while text is being written and in that case interrupting the navigation. So far I don't have a ton of faith in react-router to have this feature.

**Comment by Powersource** [comment]
Ok this actually looks pretty simple if I'm reading stuff correctly https://reactrouter.com/core/api/Prompt
